iree/compiler/Conversion/LinalgToSPIRV/ConvertToSPIRVPass.cpp

```diff
@@ -544,6 +545,16 @@ void ConvertToSPIRVPass::runOnOperation() {
   populateGPUToSPIRVPatterns(typeConverter, patterns);
   // Pull in SCF patterns to convert control flow ops.
   populateSCFToSPIRVPatterns(typeConverter, scfToSPIRVContext, patterns);
+
+  // Use the default 64-bit lowering for TOSA's ApplyScale operator:
+  //   This lowering widens integer types to 64-bit an performs the non-fused
+  //   operations, specifically multiply, add, and shift. Bit-widening
+  //   is used to guarantee higher-order bits are not truncated during the
+  //   multiply or add.
+  //
+  // TODO(antiagainst): Use a lowering that uses specific SPIRV intrinsics.
+  tosa::populateTosaRescaleToStandardConversionPatterns(&patterns);
```

Comment:
Does this lower to 64 bit floats in spirv then? `shaderFloat64` is optional (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VkPhysicalDeviceFeatures.html) so additional handling is/will be needed.

@antiagainst

Reply:
This depends on target_enviroment. If the triple and target_enviroment is set correctly, and shaderFloat64 is available. I think it will work. If not, it will use other types (maybe f32) instead.

Reply:
The LLVM compilations for SPIRV can perform simulated i64 operations by decomposing to lower bitwidths. Mul.mlir is dependent on this behavior working.